sisl/physics/electron.py

```diff
@@ -780,6 +785,9 @@ def berry_phase(contour, sub=None, eigvals=False, closed=True):
         # When the user has the contour points closed, we don't need to do this in the below loop
         closed = False
 
+    if method == 'zak':
+        closed = True
```

Comment:
Is it necessary to check that the `k[-1] - k[0] ~ 1` because otherwise the closed contour doesn't make sense?

Reply:
I think it is not necessary. If you check my snippet above one gets also the correct results by sampling just the interval `[-0.5, 0.3]`. A bit of magic here...

Reply:
Ok, but is that a coincidence or? It truly seems like magic ;)

Reply:
What is your idea about this? Do you think it generically holds, or?

Reply:
Not a coincidence, I think. It seems that the discrete formulation does not rely on having linearly-spaced contour points. And with the trick of the phase factor + going back to first point, one ends up integrating the full 1BZ.

Further, I find that the computed phases do not really change if I reduce the contour points down to only `N=4` in the snippet above. Going lower, the code fails to execute for other reasons.

In fact, in R. Resta, J. Phys.: Condens. Matter 12, R107 (2000) [http://iopscience.iop.org/article/10.1088/0953-8984/12/9/201/meta](url) points out relevance of the extreme case of _single-point Berry's phase_. This could perhaps be an interesting feature for later.

Reply:
Yes, ok, they also seem to discuss why the path need not be regularly spaced.